containerd-shim-v2/service.go

```diff
@@ -592,18 +592,18 @@ func (s *service) Pause(ctx context.Context, r *taskAPI.PauseRequest) (_ *ptypes
 	err = s.sandbox.PauseContainer(r.ID)
 	if err == nil {
 		c.status = task.StatusPaused
+		s.send(&eventstypes.TaskPaused{
+			ContainerID: c.id,
+		})
 		return empty, nil
```

Comment:
What about removing this "return" line and do a double check container status to make sure it's changed to "Paused" status, and then send the event, otherwise, return an error info?  

The following "resume" is the same case.

Reply:
IMO the `PauseContainer` will guarantee the container is paused successfully when there's no error, as it will set the container status to `StatePaused` before return (https://github.com/kata-containers/runtime/blob/stable-1.9/virtcontainers/container.go#L1261). So it should have been checked by CI rather than at runtime?

Reply:
yes, you are right.